### PR TITLE
doc: Add STM32H745I-DISCO dual-core documentation

### PR DESCRIPTION
**Pull request by nonpawite**
## Summary

Added STM32H745I-DISCO documentation for dual-core support.

### CODE REVIEW

File: Documentation/platforms/arm/stm32h7/boards/stm32h745i-disco/index.rst [hunk 1]
```diff
@@ -1,3 +1,43 @@
 ===================
 ST STM32H745I-DISCO
 ===================
+
+Dual core support
+-----------------
+
+It is recommended to use CMake to build firmware consisting of multiple images.
```
Comment (acassis) [inline Documentation/platforms/arm/stm32h7/boards/stm32h745i-disco/index.rst:8]: Is it possible to get Makefile working too, instead of CMakefile?

Comment (hartmannathan) [reply Documentation/platforms/arm/stm32h7/boards/stm32h745i-disco/index.rst:8]: This is a Documentation PR, so I think it is okay to document how things are done now. If Makefile will be able to build multi-image firmware in the future, the documentation should be updated when that happens. I prefer not to delay improvements in Documentation while waiting for Makefile support!

**Review by hartmannathan — APPROVED** [review]

**Comment by acassis** [comment]
@hartmannathan I'm just asking if Makefile is supported and why the text is saying it is recommend to use CMakefile. In theory Makefile and CMakefile needs to be kept in sync. If it is not, there is an Issue and it needs to be reported

**Comment by hartmannathan** [comment]
> @hartmannathan I'm just asking if Makefile is supported and why the text is saying it is recommend to use CMakefile. In theory Makefile and CMakefile needs to be kept in sync. If it is not, there is an Issue and it needs to be reported

@acassis Okay. Should it be reported in an issue? What to do with this PR in the meantime?

**Comment by acassis** [comment]
> > @hartmannathan I'm just asking if Makefile is supported and why the text is saying it is recommend to use CMakefile. In theory Makefile and CMakefile needs to be kept in sync. If it is not, there is an Issue and it needs to be reported
> 
> @acassis Okay. Should it be reported in an issue? What to do with this PR in the meantime?

Wait for the author to reply :-D

He opened the PR today, we need to give time for more people to review it.

**Comment by nonpawite** [comment]
Hello @acassis, sorry for the late reply. It is prefer to build with CMake because CMake can build multiple instances in different directory, which is great in dual-core case. When using Make to compile, 2 NuttX instances needs to be created which is harder compare to CMake.

**Comment by nonpawite** [comment]
Makefile is supported, but CMake is preferred!

**Comment by acassis** [comment]
> Hello @acassis, sorry for the late reply. It is prefer to build with CMake because CMake can build multiple instances in different directory, which is great in dual-core case. When using Make to compile, 2 NuttX instances needs to be created which is harder compare to CMake.

Don't worry, take your time. I think it is important to include this information there.

Did you try to compile with "make -j", how is compares to compiling with cmake ?

**Comment by nonpawite** [comment]
I haven't tried that. I will try that tomorrow since I'm not with my work machine.

**Comment by acassis** [comment]
> I haven't tried that. I will try that tomorrow since I'm not with my work machine.

Perfect! Thank you! Normally I compile this way, the compilation time drops from minutes to few seconds (5s for stm32f4discovery:nsh for instance)

**Comment by nonpawite** [comment]
Your welcome! I will let you know. @acassis

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by raiden00pl** [comment]
Using `make` to build images for AMP applications doesn't make any practical sense now, so recommending to use `cmake` makes perfect sense - compilation is usually faster and allows for out-of-tree build which makes the whole process faster and more enjoyable. It's a good practice to give users recommendations so that they don't waste their precious time :)

**Comment by nonpawite** [comment]
Hi @acassis, I have tried building with `make -j` the building speed compare to CMake is similar for me.

**Comment by acassis** [comment]
@nonpawite thanks for confirming my suspect!

**Review by acassis — APPROVED** [review]

**Pull request merged**